src/Composer/ComposerWindow.vala

```diff
@@ -37,20 +37,24 @@ public class Mail.ComposerWindow : Hdy.ApplicationWindow {
     }
 
     construct {
-        composer_widget.discarded.connect (() => {
-            close ();
-        });
-        composer_widget.sent.connect (() => {
-            close ();
-        });
-
         var titlebar = new Hdy.HeaderBar () {
             has_subtitle = false,
             show_close_button = true,
             title = _("New Message")
         };
         titlebar.get_style_context ().add_class ("default-decoration");
 
+        composer_widget.discarded.connect (() => {
+            close ();
+        });
+        composer_widget.sent.connect (() => {
+            close ();
+        });
+        composer_widget.subject_changed.connect ((subject) => {
+            bool empty_subject = subject == null || subject.length == 0;
+            titlebar.title = title = empty_subject ? _("New Message") : subject;
```

Comment:
This might be a little easier to read as:
```suggestion
            if (subject == null || subject.length == 0) {
                subject = _("New Message");
            }

            titlebar.title = title = subject
```